### PR TITLE
newdevice: Added further QNAP Turbo NAS detection #5229

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

Fixes: #5229

Adds a new option to os discover, it's a generic check (manufacturer name).

Chose to always have sysMfgName as regex compatible as I doubt people will search for the full thing here.

### CODE REVIEW

File: includes/functions.php [hunk 1]
```diff
@@ -101,8 +101,9 @@ function getHostOS($device)
 
     $sysDescr    = snmp_get($device, "SNMPv2-MIB::sysDescr.0", "-Ovq");
     $sysObjectId = snmp_get($device, "SNMPv2-MIB::sysObjectID.0", "-Ovqn");
+    $sysMfgName  = snmp_get($device, "ENTITY-MIB::entPhysicalMfgName.1", "-Ovqn");
```
Comment (murrant) [inline includes/functions.php]: Will this always be .1, can it be .0?

**Comment by murrant** [comment]
I'm unsure how I feel about this PR.  Something like this seems to be required in this case, but often using manufacturer name is obvious but often the wrong thing.  So I don't think we need to encourage it's use by making it easier.

I created a ticket with Qnap for the missing sysObjectID.

**Comment by laf** [comment]
I'm not happy either but it's the only way I could think of getting detection done without going back to doing a php file.

File: includes/functions.php (hunk 1 above)
Comment (laf) [inline includes/functions.php]: Here yes, we also use it in siklu discovery but in other vendors no it can be quit varied, Cisco seems to be [0-9]!

**Comment by murrant** [comment]
I feel like I would prefer it as a php file for now.  You don't have to remove the sysObjectId detection from the yaml file.

**Comment by laf** [comment]
Updated

**Comment by filltr** [comment]
Plase add HDD temperatures.
HDD1 - 1.3.6.1.4.1.24681.1.2.11.1.3.1
HDD2 - 1.3.6.1.4.1.24681.1.2.11.1.3.2
HDD3 - 1.3.6.1.4.1.24681.1.2.11.1.3.3
HDD4 - 1.3.6.1.4.1.24681.1.2.11.1.3.4

**Comment by NK74** [comment]
Hi @filltr , it's already mentioned here https://github.com/librenms/librenms/issues/5229#issuecomment-278596540

**Pull request merged**